src/eredis_parser.erl

```diff
@@ -135,7 +137,7 @@ do_parse({StateTag, Acc}, Data) when StateTag =:= bulk_size;
     %% this point contains the size specifier. If we cannot find it,
     %% we received a partial response and need more data
     case split_by_newline(Acc, Data) of
-        nomatch ->
+        false ->
```

Comment:
Why false rather than nomatch?

Reply:
The `nomatch` was a specific atom related to `binary:match`. Since we are no longer using `binary:match` it is misleading imo. to keep that atom.

I picked `false` because it kind of makes sense for readability -> "if split_by_newline is false then ..." reads better to me than "if split_by_newline is nomatch then ..."

Reply:
I think false is more of a boolean. Some newer OTP functions return `error` in this situation, such as `maps:find(Key, Map) -> {ok, Value} | error`.

Reply:
"if split_by_newline is false" reads like nonsense to me. :-)

Reply:
I think I saw it first in `lists:search` but I'm fine either way.
```
search(Pred, List) -> {value, Value} | false
```

Reply:
I personally don't like error when it's not really an "error" just incomplete data. Let's go with continue maybe?

Reply:
It's an error in the sense the function failed to split the data. An atom like 'notfound' is good imo but there are many variations of this. It seems {ok, Value} and {error, Reason} or just 'ok' and 'error' is the most typical erlang style nowdays. The odd stuff in the lists module has been regarded as old odd stuff. I've followed EEP discussions where this comes up, e.g. the EEP about maps.

Reply:
`continue` only describes what the return value is used for where it is currently called. The name and return values of a function should not be influenced by where it happens to be called. I don't like it.

I'm fine with either of these (but I prefer the first of them):

* `error` as in
    * `maps:find(Key, Map) -> {ok, Value} | error`
    * `dict:find(Key, Dict) -> {ok, Value} | error`
    * `dict:take(Key, Dict) -> {Value, Dict1} | error`
* `none` as in
    * `proplists:lookup(Key, List) -> none | tuple()`
    * `gb_trees:lookup(Key, Tree) -> none | {value, Value}`
* `nomatch` as in
    * `re:run(Subject, RE) -> {match, Captured} | nomatch`
    * `string:find(String, SearchPattern) -> [unicode:chardata()](https://www.erlang.org/doc/man/unicode.html#type-chardata) | nomatch`

But I don't like these:

* ~~`false`~~ (as in `lists:keysearch(Key, N, TupleList) -> {value, Tuple} | false`)
* ~~`continue`~~ ...

Reply:
`nomatch` is fine if it's not only binary:match that uses it. It also yields the minimal diff.